src/WebJobs.Script/Host/BlobLeaseManager.cs

```diff
@@ -13,26 +13,28 @@ namespace Microsoft.Azure.WebJobs.Script
 {
     internal sealed class BlobLeaseManager : IDisposable
     {
-        private const string LockBlobName = "hostlock";
-        private const string HostContainerName = "azure-functions-host";
+        internal const string LockBlobName = "hostlock";
+        internal const string HostContainerName = "azure-functions-host";
 
         private readonly Timer _timer;
         private readonly TimeSpan _leaseTimeout;
         private readonly TimeSpan _renewalInterval;
         private readonly TimeSpan _leaseRetryInterval;
         private readonly TraceWriter _traceWriter;
         private readonly string _hostId;
+        private readonly string _instanceId;
         private ICloudBlob _lockBlob;
         private string _leaseId;
         private bool _disposed;
         private bool _processingLease;
 
-        internal BlobLeaseManager(ICloudBlob lockBlob, TimeSpan leaseTimeout, string hostId, TraceWriter traceWriter, TimeSpan? renewalInterval = null)
+        internal BlobLeaseManager(ICloudBlob lockBlob, TimeSpan leaseTimeout, string hostId, string instanceId, TraceWriter traceWriter, TimeSpan? renewalInterval = null)
         {
             _lockBlob = lockBlob;
             _leaseTimeout = leaseTimeout;
             _traceWriter = traceWriter;
             _hostId = hostId;
+            _instanceId = instanceId;
 
             // Renew the lease three seconds before it expires
```

Comment:
Currently, we signal that we've lost the lease and keep retrying. Yes, this could be enhanced to have a speed up strategy leading up to the expiration time. I can create an issue for this.